presto-main/src/main/java/com/facebook/presto/sql/MaterializedViewUtils.java

```diff
@@ -175,4 +195,83 @@ public static TupleDomain<String> getDomainFromFilter(Session session, DomainTra
 
         return predicateFromBaseQuery.getTupleDomain();
     }
+
+    /**
+     * Rewrites the provided argument if it is non-associative (e.g. it cannot be handled by simply re-applying function to derived columns)
+     */
+    public static Optional<Expression> rewriteNonAssociativeFunction(FunctionCall functionCall, Map<Expression, Identifier> baseToViewColumnMap)
+    {
+        if (functionCall.isDistinct() || functionCall.getWindow().isPresent() || functionCall.getOrderBy().isPresent()
```

Comment:
instead of checking for false, should we check for expected functions?

Reply:
I don't follow, can you reword this?

Reply:
Discussed offline.